tests/integrations/virtual_environment/conftest.py

```diff
@@ -0,0 +1,60 @@
+from unittest.mock import MagicMock
+
+import pytest
+
+from briefcase.console import Console
+from briefcase.integrations.base import ToolCache
+from briefcase.integrations.virtual_environment import VenvContext
+
+
+@pytest.fixture
+def dummy_console():
+    return MagicMock(spec_set=Console)
+
+
+@pytest.fixture
+def dummy_tools(mock_tools) -> ToolCache:
+    """Override the mock_tools from parent conftest for virtual_environment tests."""
+    mock_tools.host_os = "Linux"
+
+    return mock_tools
+
+
+@pytest.fixture
+def venv_path(tmp_path):
+    return tmp_path / "test_venv"
+
+
+@pytest.fixture
+def venv_context(dummy_tools, venv_path):
+    return VenvContext(
+        tools=dummy_tools,
+        venv_path=venv_path,
+    )
+
+
+@pytest.fixture
+def mock_subprocess_setup(venv_context, monkeypatch):
+    """Setup mock objects for subprocess testing across all VenvContext tests."""
+    mock_rewrite_head = MagicMock(return_value=["rewritten", "args"])
+    mock_full_env = MagicMock(return_value={"FULL": "env"})
+    mock_subprocess = MagicMock()
+
+    mock_popen_instance = MagicMock()
+    mock_completed_process = MagicMock()
+
+    mock_subprocess.Popen.return_value = mock_popen_instance
+    mock_subprocess.run.return_value = mock_completed_process
+    mock_subprocess.check_output.return_value = "output"
+
+    monkeypatch.setattr(venv_context, "_rewrite_head", mock_rewrite_head)
+    monkeypatch.setattr(venv_context, "full_env", mock_full_env)
+    monkeypatch.setattr(venv_context.tools, "subprocess", mock_subprocess)
+
+    return {
+        "mock_rewrite_head": mock_rewrite_head,
+        "mock_full_env": mock_full_env,
+        "mock_subprocess": mock_subprocess,
+        "mock_popen_instance": mock_popen_instance,
+        "mock_completed_process": mock_completed_process,
```

Comment:
As a user affordance -everything in this dictionary is a mock - so there's no need to prefix everything with `mock_`. If this fixture was named `mock_env`, then you'd have `mock_env.rewrite_head`, `mock_env.full_env`, and so on.